test/sql/test_lead_lag/T/test_lead_lag_support_col_type

```diff
@@ -0,0 +1,37 @@
+-- name: test_lead_lag_support_col_type
```

Comment:
Please make sure that the first non-const 3-rd argument co-works with low-cardinality feature and array types in harmony.
so sql test for 3 cases as folllows are required:
1.  lead/lag(varchar_col1, n, varchar_col2), in the plan,  varchar_col adopts low-cardinalilty-dict decoding;
2. lead/lag(array_col1, n, array_col2); both array_col1 and array_col2 are array<int> or array<varchar> types;
3. lead/lag(array_col1, n, array_col2); array_col1 adopts low_cardinality-dict decoding.